### PR TITLE
[WIP, CI] codecov add Conda environment CI coverage reports

### PR DESCRIPTION
**Pull request by icfaust**
## Description

This PR adds code coverage collection and reporting from azure_pipelines LinuxCondaEnv jobs and WindowsCondaEnv jobs.  This increases overall coverage by 1% and starts covering additional sklearn versions (especially useful for ```sklearn_check_version``` and for spmd code.

No impact on performance, no benchmarking done.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
Looks like it's missing some installs:
```
2025-01-04T14:54:52.9615216Z debug - 2025-01-04 14:54:52,955 -- Selected preparation plugins --- {"selected_plugins": ["<class 'codecov_cli.plugins.xcode.XcodePlugin'>", "<class 'codecov_cli.plugins.gcov.GcovPlugin'>", "<class 'codecov_cli.plugins.pycoverage.Pycoverage'>"], "cli_config": {}}
2025-01-04T14:54:52.9616013Z debug - 2025-01-04 14:54:52,955 -- Running preparation plugin: <class 'codecov_cli.plugins.xcode.XcodePlugin'>
2025-01-04T14:54:52.9616705Z debug - 2025-01-04 14:54:52,955 -- Running xcode plugin...
2025-01-04T14:54:52.9787222Z warning - 2025-01-04 14:54:52,971 -- xcrun is not installed or can't be found.
2025-01-04T14:54:52.9788181Z debug - 2025-01-04 14:54:52,971 -- Running preparation plugin: <class 'codecov_cli.plugins.gcov.GcovPlugin'>
2025-01-04T14:54:52.9788815Z debug - 2025-01-04 14:54:52,971 -- Running gcov plugin...
2025-01-04T14:54:53.0454153Z warning - 2025-01-04 14:54:53,033 -- No gcov data found.
2025-01-04T14:54:53.0455777Z debug - 2025-01-04 14:54:53,033 -- Running preparation plugin: <class 'codecov_cli.plugins.pycoverage.Pycoverage'>
2025-01-04T14:54:53.0633472Z warning - 2025-01-04 14:54:53,049 -- coverage.py is not installed or can't be found.
2025-01-04T14:54:53.0635009Z debug - 2025-01-04 14:54:53,049 -- Collecting relevant files
2025-01-04T14:54:53.1908488Z info - 2025-01-04 14:54:53,190 -- Found 1 coverage files to report
2025-01-04T14:54:53.1911271Z info - 2025-01-04 14:54:53,190 -- > D:\a\1\s\coverage.json
2025-01-04T14:54:53.2238213Z debug - 2025-01-04 14:54:53,221 -- Selected uploader to use: <class 'codecov_cli.services.upload.upload_sender.UploadSender'>
2025-01-04T14:54:53.2377521Z debug - 2025-01-04 14:54:53,236 -- Sending upload request to Codecov
2025-01-04T14:54:53.9988611Z info - 2025-01-04 14:54:53,995 -- Process Upload complete
2025-01-04T14:54:53.9990118Z debug - 2025-01-04 14:54:53,995 -- Upload result --- {"result": "RequestResult(error=RequestError(code='HTTP Error 400', params={}, description='{\"message\":\"Repository not found\"}\\n'), warnings=[], status_code=400, text='{\"message\":\"Repository not found\"}\\n')"}
2025-01-04T14:54:53.9991030Z error - 2025-01-04 14:54:53,995 -- Upload failed: {"message":"Repository not found"}
2025-01-04T14:54:54.5197429Z ##[error]Cmd.exe exited with code '1'.
2025-01-04T14:54:54.5549346Z ##[section]Finishing: Upload to codecov
```

**Comment by icfaust** [comment]
Don't worry, thats okay. Same warnings are published on the github implementation, there are some internal conversion utilities of coverage/gcov/xcode files into proper formats which is done manually for us.

**Review by Alexsandruss — APPROVED** [review]
